src/main/java/core/basesyntax/dao/figure/FigureDaoImpl.java

```diff
@@ -3,20 +3,44 @@
 import core.basesyntax.dao.AbstractDao;
 import core.basesyntax.model.figure.Figure;
 import java.util.List;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
 
-public class FigureDaoImpl<T extends Figure> extends AbstractDao implements FigureDao<T> {
+public class FigureDaoImpl<T extends Figure> extends AbstractDao<Figure> implements FigureDao<T> {
     public FigureDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
-    public T save(T figure) {
-        return null;
+    public T save(T entity) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(entity);
+            transaction.commit();
+            return entity;
+        } catch (Exception exception) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Failed to create the entity - " + entity, exception);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<T> findByColor(String color, Class<T> clazz) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            String nameParameter = "color";
+            return session.createQuery("FROM " + clazz + " WHERE " + nameParameter + " LIKE :color", clazz)
```

Comment:
Does this query work? I cought _java.lang.IllegalArgumentException: org.hibernate.hql.internal.ast.QuerySyntaxException:_ when copy to my project